migrations/fetch-program.js

```diff
@@ -297,20 +315,105 @@ const parseTerm = (term, type, locale) => {
   return undefined;
 };
 
-const parseTag = (tag, type, locale) => {
-  if (tag) {
-    return {
-      type,
-      slug: tag.slug,
-      locales: {
-        [locale]: {
-          name: tag.name,
-          slug: tag.slug,
-        },
+const parseTags = async (tags, type, locale) => {
+  if (!tags) return undefined;
+
+  const results = [];
+  for (const tag of tags) {
+    const result = await parseTag(tag, type, locale);
+    results.push(result);
+  }
+
+  return results;
+};
+
+const parseTag = async (tag, type, locale) => {
+  if (!tag) {
+    return undefined;
+  }
+
+  const slug = parseTagSlug(tag.slug || tag.name, type);
+
+  const data = {
+    type,
+    slug: slug,
+    locales: {
+      [locale]: {
+        name: parseTagName(tag.name, type),
+        slug: slug,
       },
-    };
+    },
+  };
+
+  if (tag.icon) {
+    data.locales[locale].icon = await parseFile({ url: tag.icon });
+  }
+
+  return data;
+};
+
+const parseTagName = (name, type) => {
+  switch (type) {
+    case "activityPreparationDuration":
+      return `${name} min`;
+    default:
+      return name;
+  }
+};
+
+const parseLocationSlug = (slug) => {
+  if (!slug) return undefined;
+
+  if (slug.startsWith("A_")) {
+    return slug.slice(2).toLowerCase();
+  }
+  switch (slug) {
+    case "other":
+      return "muu";
+    case "metsä":
+      return "metsässä";
+    case "kolo":
```

Comment:
Why these 2 options translate to same "kololla" option?

Reply:
Because they their meaning was identical in the old data:

```json
{ "name": "Kololla", "slug": "A_Kololla" },
```

and 
```json
{ "name": "Kolo", "slug": "meeting_place" },
```